### PR TITLE
[RLlib] Model Annotations to Torch Models 

### PR DESCRIPTION
**Pull request by michaelzhiluo**
## Why are these changes needed?

No annotations for torch models.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [x] I've included any doc changes needed for https://docs.ray.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

File: rllib/models/preprocessors.py [hunk 1]
```diff
@@ -106,7 +106,7 @@ def _init_shape(self, obs_space, options):
         return shape
 
     @override(Preprocessor)
-    def transform(self, observation):
+    def transform(self, observation: Any) -> np.ndarray:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:109]: Shouldn't this be TensorType?

File: rllib/models/preprocessors.py [hunk 2]
```diff
@@ -130,50 +130,50 @@ def transform(self, observation):
 
 class AtariRamPreprocessor(Preprocessor):
     @override(Preprocessor)
-    def _init_shape(self, obs_space, options):
+    def _init_shape(self, obs_space: gym.Space, options: dict) -> List[int]:
         return (128, )
 
     @override(Preprocessor)
-    def transform(self, observation):
+    def transform(self, observation: Any) -> np.ndarray:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:137]: TensorType

File: rllib/models/preprocessors.py [hunk 3]
```diff
@@ -130,50 +130,50 @@ def transform(self, observation):
...
 
 class OneHotPreprocessor(Preprocessor):
     @override(Preprocessor)
-    def _init_shape(self, obs_space, options):
+    def _init_shape(self, obs_space: gym.Space, options: dict) -> List[int]:
         return (self._obs_space.n, )
 
     @override(Preprocessor)
-    def transform(self, observation):
+    def transform(self, observation: Any) -> np.ndarray:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:148]: TensorType

File: rllib/models/preprocessors.py [hunk 4]
```diff
@@ -130,50 +130,50 @@ def transform(self, observation):
...
         self.check_shape(observation)
         arr = np.zeros(self._obs_space.n, dtype=np.float32)
         arr[observation] = 1
         return arr
 
     @override(Preprocessor)
-    def write(self, observation, array, offset):
+    def write(self, observation: Any, array: np.ndarray, offset: int) -> None:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:155]: TensorType

File: rllib/models/preprocessors.py [hunk 5]
```diff
@@ -130,50 +130,50 @@ def transform(self, observation):
...
 
 class NoPreprocessor(Preprocessor):
     @override(Preprocessor)
-    def _init_shape(self, obs_space, options):
+    def _init_shape(self, obs_space: gym.Space, options: dict) -> List[int]:
         return self._obs_space.shape
 
     @override(Preprocessor)
-    def transform(self, observation):
+    def transform(self, observation: Any) -> np.ndarray:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:165]: TensorType

File: rllib/models/preprocessors.py [hunk 6]
```diff
@@ -130,50 +130,50 @@ def transform(self, observation):
...
         self.check_shape(observation)
         return observation
 
     @override(Preprocessor)
-    def write(self, observation, array, offset):
+    def write(self, observation: Any, array: np.ndarray, offset: int) -> None:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:170]: TensorType

File: rllib/models/preprocessors.py [hunk 7]
```diff
@@ -197,14 +197,14 @@ def _init_shape(self, obs_space, options):
         return (size, )
 
     @override(Preprocessor)
-    def transform(self, observation):
+    def transform(self, observation: Any) -> np.ndarray:
```
Comment (sven1977) [inline rllib/models/preprocessors.py:200]: TensorType

File: rllib/models/torch/misc.py [hunk 8]
```diff
@@ -84,20 +88,20 @@ def __init__(
...
-                 initializer=None,
-                 activation_fn=None,
-                 use_bias=True,
-                 bias_init=0.0):
+                 in_size: int,
+                 out_size: int,
+                 initializer: Any = None,
+                 activation_fn: Any = None,
+                 use_bias: bool = True,
+                 bias_init: float = 0.0):
```
Comment (sven1977) [inline rllib/models/torch/misc.py:104]: Could you add a docstring here?

File: rllib/models/torch/modules/noisy_layer.py [hunk 9]
```diff
@@ -16,7 +17,11 @@ class NoisyLayer(nn.Module):
     vanish along the training procedure
     """
 
-    def __init__(self, in_size, out_size, sigma0, activation="relu"):
+    def __init__(self,
+                 in_size: int,
+                 out_size: int,
+                 sigma0: float,
+                 activation: str = "relu"):
```
Comment (sven1977) [inline rllib/models/torch/modules/noisy_layer.py:24]: Sorry, the incomplete docstring here is on me. Could you complete it, though? Thanks.

**Review by sven1977 — CHANGES_REQUESTED** [review]
Only minor stuff.
- TensorType instead of Any
- some docstrings to be added

File: rllib/models/preprocessors.py (hunk 1 above)
Comment (michaelzhiluo) [reply rllib/models/preprocessors.py:109]: The base class PreProcessor has np.ndarray as typing. I am assuming this applies to all other subclasses

File: rllib/models/preprocessors.py (hunk 4 above)
Comment (michaelzhiluo) [reply rllib/models/preprocessors.py:155]: Similar, refer to the comment above

File: rllib/models/preprocessors.py (hunk 1 above)
Comment (michaelzhiluo) [reply rllib/models/preprocessors.py:109]: @sven1977

**Comment by michaelzhiluo** [comment]
@sven1977 It should be ready to merge!

**Review by sven1977 — APPROVED** [review]
LGTM, let's get this merged.

**Comment by sven1977** [comment]
@michaelzhiluo, could you merge master into this branch once more?

**Pull request merged**